src/robot.cpp

```diff
@@ -641,6 +655,28 @@ geometry_msgs::Pose Robot::getManipulationTarget()
   return current_pose;  
 }
 
+std::vector<std::string> Robot::getNamedTargetPoses(std::string& planning_group_name)
```

Comment:
like mentioned above, use const ref. i do see that you are doing some planing group name retrieval logic, but in that case just create a local variable for it. we don't want to actually return anything via the `planning_group_name` argument right?

Reply:
Ok, 
Added const to the planning_group_name argument